src/commands.ts

```diff
@@ -23,35 +24,50 @@ import * as vscode from 'vscode';
  *   https://code.visualstudio.com/api/extension-guides/command
  */
 
-/**
+// flags to indicating whether a resolve or update is in progress
+var resolveRunning = false;
```

Comment:
Do we need these to be globals, or can we move them into `commands` as properties?

Reply:
It wasn't happy having them inside `commands`. Commands is not a class

Reply:
We can upgrade it to a class and just export a singleton object, like what I did with the `StatusItem`? That would allow us to make these flags `private`.

Reply:
I upgraded it to a class and made all the members static, so we don't need to drag a singleton around all over the place